org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/JavaSearchScope.java

```diff
@@ -273,9 +274,11 @@ public void add(IJavaElement element) throws JavaModelException {
 			// remember sub-cu (or sub-class file) java elements
 			if (element instanceof IMember) {
 				if (this.elements == null) {
-					this.elements = new ArrayList();
+					this.elements = new HashSet();
+					this.elementTypes = new boolean[IJavaElement.JAVA_MODULE+1];
```

Comment:
I'd rather use a BitSet which would grow dynamically. Otherwise it's hard to remember that the boolean array-size must strictly depend on the largest integer constant from IJavaElement.

Reply:
@eric-milles ping

Reply:
The PR is good as-is IMO.  It is in-keeping with the style of the existing source.  You can refactor `JavaSearchScope` after if you wish.

Reply:
The chances of a merge are higher if you just follow reviewer comments. Typically no one takes time to refactor code after a merge.